src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/PrimitiveDataContract.cs

```diff
@@ -4,28 +4,46 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 using System.Reflection;
+using System.Runtime.Serialization.Json;
 using System.Xml;
 
 namespace System.Runtime.Serialization
 {
     internal abstract class PrimitiveDataContract : DataContract
     {
-        internal static readonly PrimitiveDataContract NullContract = new NullPrimitiveDataContract();
+        internal static PrimitiveDataContract? s_NullContract;
+        internal static PrimitiveDataContract NullContract
+        {
+            [RequiresUnreferencedCode(DataContractJsonSerializer.SerializerTrimmerWarning)]
```

Comment:
What about the NullPrimitiveDataContract requires unreferenced code? Seems like we should be able to suppress the warning at the source, since this is the "null" primitive data contract.

Reply:
It seems like it would be a lot less changes if `NullPrimitiveDataContract` suppressed the warning itself. We wouldn't need to make any changes here. Nor in the `DataMember.CriticalHelper` ctor:

```
            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
               Justification = "The only use of this field is on the following property getter which is already " +
                "annotated as RequiresUnreferencedCode.")]
            internal CriticalHelper(MemberInfo memberInfo)
            {
                _emitDefaultValue = Globals.DefaultEmitDefaultValue;
                _memberInfo = memberInfo;
                _memberPrimitiveContract = PrimitiveDataContract.NullContract;
            }
```

Reply:
> and suppress the IL2046 errors

I think just suppress it in the NullPrimitiveDataContract ctor. Keep the overrides as `RequiresUnreferencedCode`, since those are never called.